browser-interface/packages/shared/session/index.ts

```diff
@@ -1,79 +1,92 @@
+  await saveToPersistentStorage(sessionKey(session.identity.address), session)
+}
+
+/**
+ * Forget a session existing in the current PersistentStorage
+ */
+export const deleteSession = async (userId?: string) => {
+  if (userId) {
+    await removeFromPersistentStorage(sessionKey(userId))
+  }
+}
+
+/**
+ * Retrieve the last session stored by address
+ */
+export const retrieveLastSessionByAddress: (address: string) => Promise<StoredSession | null> = async (
+  address: string
+) => {
+  const sessions = (await retrieveAllCurrentSessions()).filter(withAddress(address))
+
+  return sessions.length > 0 ? sessions[0] : null
+}
+
+/**
+ * Retrieve the last session stored that is a guest session
+ */
+export const retrieveLastGuestSession: () => Promise<StoredSession | null> = async () => {
+  const sessions = (await retrieveAllCurrentSessions()).filter(isGuest).sort(byExpiration).sort()
+
+  return sessions.length > 0 ? sessions[0] : null
+}
+
+/**
+ * Get the current provider (utility function that calls store.getState())
+ */
+export const getProvider = () => getProviderSelector(store.getState())
+
 const SESSION_KEY_PREFIX = 'dcl-session'
 const LAST_SESSION_KEY = 'dcl-last-session-id'
 
 function sessionKey(userId: string) {
   return `${SESSION_KEY_PREFIX}-${userId.toLocaleLowerCase()}`
 }
 
-export const setStoredSession: (session: StoredSession) => Promise<void> = async (session) => {
-  await saveToPersistentStorage(LAST_SESSION_KEY, session.identity.address)
-  await saveToPersistentStorage(sessionKey(session.identity.address), session)
+async function retrieveAllCurrentSessions(): Promise<StoredSession[]> {
+  const allKeys = await getKeysFromPersistentStorage()
+  const sessionKeys = allKeys.filter((_) => _.startsWith(SESSION_KEY_PREFIX))
+  const sessions = await Promise.all(sessionKeys.map(getFromPersistentStorage))
+  return sessions.filter(notNull).filter(notExpired)
 }
 
-export const getStoredSession: (userId: string) => Promise<StoredSession | null> = async (userId) => {
-  const existingSession: StoredSession | null = (await getFromPersistentStorage(sessionKey(userId))) as StoredSession
-
-  if (existingSession) {
-    return existingSession
-  } else {
-    // If not existing session was found, we check the old session storage
-    const oldSession: StoredSession | null = (await getFromPersistentStorage('dcl-profile')) || {}
-    if (oldSession && oldSession.identity && oldSession.identity.address === userId) {
-      await setStoredSession(oldSession)
-      return oldSession
-    }
-  }
-
-  return null
+function isGuest(session: StoredSession) {
+  return !!session.isGuest
 }
 
-export const removeStoredSession = async (userId?: string) => {
-  if (userId) await removeFromPersistentStorage(sessionKey(userId))
+function notNull(session: StoredSession | null): session is StoredSession {
+  return !!session
 }
-export const getLastSessionWithoutWallet: () => Promise<StoredSession | null> = async () => {
```

Comment:
This is not used anywhere: https://sourcegraph.com/search?q=context:%40eordano/decentraland+getLastSessionWithoutWallet&patternType=standard&sm=1&groupBy=repo